vowpalwabbit/loss_functions.cc

```diff
@@ -292,6 +322,58 @@ class quantileloss : public loss_function
   float tau;
 };
 
+class expectileloss : public loss_function
+{
+public:
+  expectileloss(float& q_) : q(q_) {}
+
+  std::string getType() const override { return "expectile"; }
+  float getParameter() const override { return q; }
+
+  // Expectile loss is closely related to the squared loss, but it's an assymetric function with a quantile parameter
+  // Its methods can be derived from the corresponding methods from the squared loss multiplied by the quantile value
+  float getLoss(const shared_data* sd, float prediction, float label) const override
+  {
+    float err = label - prediction;
+    return squared_loss_impl::getLoss(sd, prediction, label) * (err > 0 ? q : (1.f - q));
+  }
+
+  float getUpdate(float prediction, float label, float update_scale, float pred_per_update) const override
+  {
+    // Based on the invariant update calculation from the [Online Importance Weight Aware Updates paper](https://arxiv.org/abs/1011.1576)
```

Comment:
This comment is not specific to expectile loss, maybe remove or move to header?

Reply:
I agree, moved it to the header file.